openpype/hosts/maya/api/lib.py

```diff
@@ -1737,8 +1737,11 @@ def apply_shaders(relationships, shadernodes, nodes):
             log.warning("No nodes found for shading engine "
                         "'{0}'".format(id_shading_engines[0]))
             continue
+        try:
+            cmds.sets(filtered_nodes, forceElement=id_shading_engines[0])
+        except RuntimeError as rte:
+            log.error("Error during shader assignment: {}".format(rte))
```

Comment:
Could you describe the errors you were facing that needed this to solve it?

Reply:
It was a case we had with outsourcing files, where the 'initialShadingGroup' node was locked. 
```
# Traceback (most recent call last):
#   File "D:\dev\OpenPype\openpype\tools\mayalookassigner\app.py", line 253, in on_process_selected
#     assign_look_by_version(nodes, version_id=version["_id"])
#   File "D:\dev\OpenPype\openpype\hosts\maya\api\lib.py", line 1608, in assign_look_by_version
#     apply_shaders(relationships, shader_nodes, nodes)
#   File "D:\dev\OpenPype\openpype\hosts\maya\api\lib.py", line 1724, in apply_shaders
#     cmds.sets(filtered_nodes, forceElement=id_shading_engines[0])
# RuntimeError: # Connection not made: 'wesley_shield_03_geoShape.instObjGroups[0]' -> 'initialShadingGroup.dagSetMembers[88]'.  Destination is locked.
Error while parsing arguments.
# 
```

The point of this change is so that the whole process won't abort the first time there's an error. If it happens that the first piece of geo gives an error, Look assigner probably still should try for the rest.